### PR TITLE
Replace YUM and migrate to Python 3

### PR DESCRIPTION
**Pull request by mkutlak**
This PR builds on top of the PR: https://github.com/abrt/retrace-server/pull/97

Changes in this PR:
- replace `YUM` with `DNF`
- replace `createrepo` with `createrepo_c`
- additional changes to Python3 compatibility (strings/bytes, raw strings)
- migrate retrace-server to Python3

And because these changes are not compatible with RHEL7 there is a new branch:
https://github.com/abrt/retrace-server/tree/rhel7

### CODE REVIEW

File: src/retrace/retrace.py [hunk 1]
```diff
@@ -18,8 +18,9 @@
 from subprocess import *
 import magic
 import six
+from dnf.subject import Subject
+from  hawkey import FORM_NEVRA
```
Comment (marusak) [inline src/retrace/retrace.py]: isn't hawkey obsoleted?

File: src/retrace/retrace.py [hunk 2]
```diff
@@ -427,6 +428,23 @@ def run_gdb(savedir, plugin):
...
+    Return a name, version, release, epoch, arch, e.g.::
+        foo-1.0-1.i386.rpm returns foo, 1.0, 1, i386
+    """
+
+    if filename[-4:] == '.rpm':
+        filename = filename[:-4]
+
+    subject = Subject(filename)
+    possible_nevra = subject.get_nevra_possibilities(forms=FORM_NEVRA)
+    nevra = next(iter(possible_nevra))
```
Comment (marusak) [inline src/retrace/retrace.py]: Test for None would be nice

File: src/retrace/retrace.py (hunk 1 above)
Comment (mkutlak) [reply src/retrace/retrace.py]: Yes, hawkey as a project is obsolete, but it is part of libdnf.

File: src/retrace/retrace.py (hunk 2 above)
Comment (mkutlak) [reply src/retrace/retrace.py]: Done.

File: src/retrace-server-reposync-faf [hunk 3]
```diff
@@ -20,30 +21,87 @@ def get_pkglist(db, outputdir, opsys, release, arch):
...
+    fil_xml = cr.FilelistsXmlFile(fil_xml_path)
+    oth_xml = cr.OtherXmlFile(oth_xml_path)
+    pri_db = cr.PrimarySqlite(pri_db_path)
+    fil_db = cr.FilelistsSqlite(fil_db_path)
+    oth_db = cr.OtherSqlite(oth_db_path)
+
+    # Prepare list of packages to process
+    pkg_list = get_pkglist(db, outputdir, opsys, release, arch)
+
+    pri_xml.set_num_of_pkgs(len(pkg_list))
```
[the remark this answers is not in the document]
Comment (mkutlak) [reply src/retrace-server-reposync-faf]: Fixed.

**Pull request merged**